File: TypeScript/Codewars/src/resolutions/Find-the-next-perfect-square.ts
export function findNextSquare(sq: number): number {
	const squareRoot = Math.sqrt(sq);

	const isPerfectSquareRoot = squareRoot === Number.parseInt(squareRoot.toString());

	return isPerfectSquareRoot ? (squareRoot + 1) ** 2 : -1;
}

File: Csharp/CodeWars/src/Completed/FindTheNextPerfectSquare.cs
namespace CodeWars.Completed;

public class FindTheNextPerfectSquare
{
    public static long FindNextSquare(long num)
    {
        double squareRoot = Math.Sqrt(num);
        if (squareRoot % 1 != 0)
            return -1;

        long root = (long)squareRoot;
        if (root * root != num)
            return -1;

        long next = root + 1;
        return next * next;
    }
}
